### DISCUSSION TITLE
Modify the size of the mesh region

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#4755] [Q&A] [answered]
Hi, In the process of building a thermal simulation model, I am using the pyaedt script, which gives me a positive loop feedback to my work.
I am using the following script to assign mesh region.

```
mesh_object_list = ipk.modeler.solid_names
mesh_object_list.remove("Region")
mesh_region = ipk.mesh.assign_mesh_region(objectlist=mesh_objectlist)
```

But there is a small problem with this. This problem is that the mesh region fits tightly to the solid as shown below.

![image](https://github.com/ansys/pyaedt/assets/163245306/f424331f-134a-46bc-9334-b6c7c01e7536)

My requirement is to increase the size of the mesh region slightly, which is good for the mesh quality. In terms of specific parameter changes, firstly, the X, Y, and Z positions are each reduced by 0.1mm. Secondly, XSize, YSize, and ZSize are each increased by 0.2mm.

![image](https://github.com/ansys/pyaedt/assets/163245306/c5924b52-0ca3-493a-8fb0-51a41756ee15)

These parameter changes achieve the results I want as shown below.

![image](https://github.com/ansys/pyaedt/assets/163245306/39411b51-ad73-407a-870f-f82002bd0ec9)

The version of pyaedt I am using is 0.9.2. I would be very happy to know what script would enable the modification of the mesh region size. Looking forward to your reply!

**Comment by lorenzovecchietti** [comment, marked as answer]
You can change settings from the mesh_region object.
You can switch between manual or automatic mesh settings using:
```
mesh_region.manual_settings = True
mesh_region.manual_settings = False
```

To print the settings available:
```
mesh_region.settings
```

That line will print out one of the two dictionaries:
- `{'EnforceCutCellMeshing': False, 'StairStepMeshing': False, 'OptimizePCBMesh': True, 'CurvatureSizeFunction': True, 'EnableTransition': False, 'Enforce2dot5DCutCell': False, 'Enable2DCutCell': False, 'ProximitySizeFunction': True, 'MeshRegionResolution': 3}`
- `{'2DMLMType': '2DMLM_None', 'EnableMLM': True, 'EnforceCutCellMeshing': False, 'MinGapX': '1mm', 'NoOGrids': False, 'MaxElementSizeY': '0.02mm', 'StairStepMeshing': False, 'UniformMeshParametersType': 'Average', 'MinGapZ': '1mm', 'CurvatureSizeFunction': True, 'EnableTransition': False, 'MaxElementSizeX': '0.02mm', 'EnforeMLMType': '3D', 'MinElementsInGap': '3', 'MinElementsOnEdge': '2', 'OptimizePCBMesh': True, 'BufferLayers': '0', 'MaxElementSizeZ': '0.03mm', 'MaxSizeRatio': '2', 'MaxLevels': '0', 'MinGapY': '1mm', 'Enforce2dot5DCutCell': False, 'Enable2DCutCell': False, 'ProximitySizeFunction': True}`

As an example:
```
mesh_region.settings["MaxElementSizeX"] = "2mm"
mesh_region.settings["MaxElementSizeY"] = "2mm"
mesh_region.settings["MaxElementSizeZ"] = "2mm"
mesh_region.settings["EnableMLM"] = True
mesh_region.settings["MaxLevels"] = "2"
mesh_region.settings["MinElementsInGap"] = 4
mesh_region.update()
```

**Reply by aknsj** [reply]
Thanks for your reply, I can successfully change the mesh element size by mesh_region.setting, such as “MaxElementSizeX”. But I still want to know which parameter can change the mesh region size, thanks!

**Reply by lorenzovecchietti** [reply]
You should be able to do something like:
```
mesh_region.assignment. positive_x_padding_type = "Absolute Position"
mesh_region.assignment. positive_x_padding = "3mm"
```

See all the properties of the subregion class from the documentation or from here: https://github.com/ansys/pyaedt/blob/99f7471f129837ec36a60b8ede88545a239f59c4/pyaedt/modules/MeshIcepak.py#L17

**Reply by aknsj** [reply]
Thanks for your reply.I used the following code to assign the mesh region.
```
mesh_objectlist = ipk.modeler.solid_names
mesh_objectlist.remove("Region")
mesh_region = ipk.mesh.assign_mesh_region(assignment=mesh_objectlist)
mesh_region.manual_settings = True
mesh_region.settings["MaxElementSizeX"] = "3mm"
mesh_region.settings["MaxElementSizeY"] = "3mm"
mesh_region.settings["MaxElementSizeZ"] = "1mm"
mesh_region.settings["MinGapX"] = "0.01mm"
mesh_region.settings["MinGapY"] = "0.01mm"
mesh_region.settings["MinGapZ"] = "0.01mm"
mesh_region.settings["MaxLevels"] = "3"
mesh_region.settings["MinElementsInGap"] = 1
mesh_region.settings["EnableMLM"] = True
mesh_region.update()
```
But while modifying the mesh region size using the following code, the following error occurs.
```
mesh_region.assignment.positive_x_padding_type = "Absolute Position"
mesh_region.assignment.positive_x_padding = "3mm"
```
```
err: 'list' object has no attribute 'positive_x_padding_type'
```

**Reply by lorenzovecchietti** [reply]
Hi, what AEDT version are you using?
The procedure should work on 24R1. 
If you are using an old version, please try to access directly the non-model object that defines the mesh region:
```
region_box=icepak.modeler["Box2"]
h=region_box.history()
h.props["XSize"]="10mm"
```

**Reply by aknsj** [reply]
Hi, I was previously using AEDT version 2022R2, I upgraded to version 2024R1 and tested these scripts below. I successfully changed the mesh region size. Thanks a lot!
```
mesh_region.assignment.positive_x_padding_type = "Absolute Position"
mesh_region.assignment.positive_x_padding = "3mm"
```
